ddtrace/llmobs/_integrations/langgraph.py

```diff
@@ -60,33 +77,53 @@ def maybe_format_langchain_io(messages):
                 INPUT_VALUE: format_langchain_io(inputs),
                 OUTPUT_VALUE: maybe_format_langchain_io(response)
                 or maybe_format_langchain_io(span._get_ctx_item(LANGGRAPH_ASTREAM_OUTPUT)),
-                NAME: self._graph_nodes_by_task_id.get(instance_id, {}).get("name") or kwargs.get("name", span.name),
+                NAME: invoked_node.get("name") or kwargs.get("name", span.name),
                 SPAN_LINKS: current_span_links + span_links,
             }
         )
-        if operation == "graph" and not _is_subgraph(span):
-            self._graph_nodes_by_task_id.clear()
+
+    def _get_node_metadata_from_span(self, span: Span, instance_id: str) -> Dict[str, Any]:
+        """
+        Get the node metadata for a given span and its node instance id.
+        Additionally, set the span and trace ids on its metadata should another node in a later
+        tick of the graph need to be linked to this node.
+        """
+        parent_span = _get_nearest_llmobs_ancestor(span)
+        invoked_node = (
+            self._graph_nodes_for_graph_by_task_id.setdefault(parent_span, {}).setdefault(instance_id, {})
+            if parent_span
+            else {}
+        )
+        invoked_node["span"] = {"trace_id": format_trace_id(span.trace_id), "span_id": str(span.span_id)}
+        return invoked_node
 
     def llmobs_handle_pregel_loop_tick(
         self, finished_tasks: dict, next_tasks: dict, more_tasks: bool, is_subgraph_node: bool = False
     ):
-        """Compute incoming and outgoing span links between finished tasks and queued tasks in the graph."""
-        if not self.llmobs_enabled:
-            return
-        graph_span = (
-            LLMObs._instance._current_span()
-        )  # we're running between nodes, so the current span should be the pregel graph
-        if graph_span is None:
-            return
+        """
+        Compute incoming and outgoing span links between finished tasks and queued tasks in the graph.
 
-        if not more_tasks:
-            self._handle_finished_graph(graph_span, finished_tasks, is_subgraph_node)
-            return
-        finished_task_names_to_ids = {task.name: task_id for task_id, task in finished_tasks.items()}
-        for task_id, task in next_tasks.items():
-            self._link_task_to_parent(task_id, task, finished_task_names_to_ids)
+        In the case of finished and queued tasks, any finished tasks that aren't used as links to
+        queued tasks should be linked to the encompassing graph span's output.
```

Comment:
```suggestion
```
This is already addressed in the `_handle_intermediary_graph_tick` docstring I think